homeassistant/components/heos/__init__.py

```diff
@@ -168,31 +168,24 @@ def __init__(self, hass, controller):
         self._device_registry = None
         self._entity_registry = None
         self.controller = controller
-        self._signals = []
 
     async def connect_listeners(self):
         """Subscribe to events of interest."""
         self._device_registry = dr.async_get(self._hass)
         self._entity_registry = er.async_get(self._hass)
 
         # Handle controller events
-        self._signals.append(
-            self.controller.dispatcher.connect(
-                heos_const.SIGNAL_CONTROLLER_EVENT, self._controller_event
-            )
+        self.controller.dispatcher.connect(
```

Comment:
The previous code was cleaning this up in disconnect since it was adding to _signals. Is it correct that this is unnecessary now?

Reply:
Correct, it's only adding to the HEOS dispatcher and calling disconnect_all removes them all, so there's no need to capture the individual disconnects.

Reply:
Great question @dcmeglio !